src/gordon_gcp/plugins/service/gdns_publisher.py

```diff
@@ -275,18 +271,19 @@ def _format_resource_record_changes(self, action, resource_record):
 
         return changes
 
-    async def _changes_published(self, change_id):
+    async def _changes_published(self, change_id, base_changes_url):
         """Checks if a given Change resource has a completed status.
 
         See DNS_CHANGES_DONE.
 
         Args:
             change_id (str): Change ID of the Changes resource.
+            base_changes_url (str): The URL prefix to use to check on changes.
 
         Returns:
             True if the changes are done, False if anything else.
         """
-        change_id_url = f'{self.base_changes_url}/{change_id}'
```

Comment:
Why not pass the zone name instead of urls? Then you can just do the formatting once, rather than twice.

Reply:
You mean do it in this function?  The base_changes_url is used twice, so I didn't want to duplicate the code.  I suppose the resource_records_url is only used once, so it might make sense there.

Reply:
We talked about this outside of the PR and decided doing a minimal change (i.e. just the url) was better than a larger refactor for now.